contracts/lbp/LBPManager.sol

```diff
@@ -0,0 +1,290 @@
+     * @param _userData                 UserData object that specifies the type of join.
+     */
+    function addLiquidity(
+        IERC20[] memory _tokenList,
+        uint8 _projectTokenIndex,
+        address _sender,
+        bytes memory _userData
+    ) external onlyAdmin {
+        require(!poolFunded, "LBPManager: pool has already been funded");
+        poolFunded = true;
+
+        IVault vault = lbp.getVault();
+
+        if (primeDaoFeePercentage != 0) {
+            // Transfer primeDaoFee to beneficiary.
+            _tokenList[_projectTokenIndex].transferFrom(
+                _sender,
+                beneficiary,
+                _feeAmountRequired(_projectTokenIndex)
+            );
+        }
+
+        for (uint8 i; i < _tokenList.length; i++) {
+            _tokenList[i].transferFrom(_sender, address(this), amounts[i]);
+            _tokenList[i].approve(address(vault), amounts[i]);
+        }
+
+        IVault.JoinPoolRequest memory request = IVault.JoinPoolRequest({
+            maxAmountsIn: amounts,
+            userData: _userData,
+            fromInternalBalance: false, // It is not possible to add liquidity through the internal Vault balance.
+            assets: _tokenList
+        });
+
+        vault.joinPool(lbp.getPoolId(), address(this), address(this), request);
+    }
+
+    /**
+     * @dev                             exit pool or remove liquidity from pool
+     * @param _tokenList                Numerically sorted array (ascending) containing two addresses:
+                                            - The address of the project token being distributed.
+                                            - The address of the funding token being exchanged for the project token.
+     * @param _receiver                 Address of the liquidity receiver, after exiting the LBP.
+     * @param _toInternalBalance        Balancer's Vault option to send the liquidity from the LBP to the _receivers internal Vault balance.
+     * @param _userData                 UserData object that specifies the type of exit.
+     */
+    function removeLiquidity(
+        IERC20[] memory _tokenList,
+        address payable _receiver,
+        bool _toInternalBalance,
+        bytes memory _userData
+    ) external {
+        require(
+            _receiver != payable(address(0)),
+            "LBPManager: receiver of project and funding tokens can't be zero"
+        );
+        require(
+            lbp.balanceOf(address(this)) > 0,
+            "LBPManager: wrapper does not have any pool tokens to remove liquidity"
+        );
+
+        uint256 endTime;
+        (, endTime, ) = lbp.getGradualWeightUpdateParams();
+
+        require(
+            block.timestamp >= endTime,
+            "LBPManager: can not remove liqudity from the pool before endtime"
+        );
+
+        // To remove all funding from the pool. Initializes to [0, 0]
+        uint256[] memory _minAmountsOut = new uint256[](2);
```

Comment:
Btw, why the array length here is exactly 2?

Reply:
Because Balancers LBP only functions with two tokens. But good point, I made the size dynamic